tests/remote_point_evaluation/vector_tools_evaluate_at_points_01.cc

```diff
@@ -0,0 +1,253 @@
+
+void
+test()
+{
+  const unsigned int dim             = 2;
+  const unsigned int degree          = 2;
+  const unsigned int n_refinements_1 = 3;
+  const unsigned int n_refinements_2 = 5;
+
+  parallel::distributed::Triangulation<dim> tria_1(MPI_COMM_WORLD);
+#if true
+  GridGenerator::hyper_cube(tria_1, -1.0, +1.0);
+  tria_1.refine_global(n_refinements_1);
+#else
+  GridGenerator::create_annulus(tria_1, n_refinements_2);
+#endif
+  DoFHandler<dim> dof_handler_1(tria_1);
+  FE_Q<dim>       fe_1(degree);
+  dof_handler_1.distribute_dofs(fe_1);
+
+  parallel::distributed::Triangulation<dim> tria_2(MPI_COMM_WORLD);
+#if true
+  GridGenerator::create_annulus(tria_2, n_refinements_2);
+#else
+  GridGenerator::hyper_cube(tria_2, -1.0, +1.0);
+  tria_2.refine_global(n_refinements_1);
+#endif
+  DoFHandler<dim>           dof_handler_2(tria_2);
+  FE_DGQArbitraryNodes<dim> fe_2(QGauss<1>(degree + 1).get_points());
+  dof_handler_2.distribute_dofs(fe_2);
+
+  LinearAlgebra::distributed::Vector<double> vector_1(
+    create_partitioner(dof_handler_1));
+
+  const MappingQ1<dim> mapping_1;
+  VectorTools::interpolate(mapping_1,
+                           dof_handler_1,
+                           AnalyticalFunction<dim>(),
+                           vector_1);
+
+  const MappingQ1<dim> mapping_2;
+  const QGauss<dim>    quadrature_2(degree + 1);
+  FEValues<dim>        fe_values_2(mapping_2,
+                            fe_2,
+                            quadrature_2,
+                            update_quadrature_points);
+
+  std::vector<Point<dim>> evaluation_points;
+  for (const auto &cell : dof_handler_2.active_cell_iterators())
+    {
+      if (cell->is_locally_owned() == false)
+        continue;
+
+      fe_values_2.reinit(cell);
+      evaluation_points.insert(evaluation_points.end(),
+                               fe_values_2.get_quadrature_points().begin(),
+                               fe_values_2.get_quadrature_points().end());
+    }
+
+  vector_1.update_ghost_values();
+  Utilities::MPI::RemotePointEvaluation<dim> evaluation_cache;
+  const auto evaluation_point_results = VectorTools::evaluate_at_points<1>(
+    mapping_1, dof_handler_1, vector_1, evaluation_points, evaluation_cache);
+  vector_1.zero_out_ghosts();
+
+  LinearAlgebra::distributed::Vector<double> vector_2(
+    create_partitioner(dof_handler_2));
+
+  auto ptr = evaluation_point_results.begin();
+  for (const auto &cell : dof_handler_2.active_cell_iterators())
+    {
+      if (cell->is_locally_owned() == false)
+        continue;
+
+      fe_values_2.reinit(cell);
+
+      Vector<double> temp(ptr, ptr + fe_values_2.n_quadrature_points);
+
+      cell->set_dof_values(temp, vector_2);
+
+      ptr += fe_values_2.n_quadrature_points;
+    }
+
+  if (false)
+    {
+      print(mapping_1, dof_handler_1, vector_1, 0);
+      print(mapping_2, dof_handler_2, vector_2, 1);
+    }
+
+  vector_1.print(deallog.get_file_stream());
+  vector_2.print(deallog.get_file_stream());
```

Comment:
How do you check for correctness here?

Reply:
I took a look at the result in Paraview. If you want I can replace the printing of the vectors by computing the global error for the following tests: `vector_tools_evaluate_at_points_01`, `vector_tools_evaluate_at_points_03`, `vector_tools_evaluate_at_points_04`, `vector_tools_evaluate_at_points_05`. For the other tests, it is - as far as I know - not possible to come up with some derived quantities.

Reply:
Yes, I would prefer to check for the global error in the tests you mentioned.

Reply:
After that, it's good from my side.

Reply:
OK. Done! Thanks again for the review!